client/app/src/main/java/com/example/criengine/FilterFragmentActivity.java

```diff
@@ -122,4 +108,19 @@ public void onClick(DialogInterface dialogInterface, int i) {
                     }})
                 .create();
     }
+
+    /**
+     * Adds the requests status to the list. This eventually determines what the view wants to
+     * view.
+     * @param checkBox The check box view.
+     * @param isChecked Whether the check box is currently active.
+     */
+    @Override
+    public void onCheckedChanged(CompoundButton checkBox, boolean isChecked) {
+        if (isChecked) {
+            filterStatus.add(checkBox.getText().toString().toLowerCase());
+        } else {
+            filterStatus.remove(checkBox.getText().toString().toLowerCase());
+        }
+    }
 }
```

Comment:
kinda feels sketch that you're using the checkbox text where it's dependent on the xml. I feel like this should be set somewhere in the code to guarantee

Reply:
its either using the name which *SHOULD* (and is) be identical to the status of the books or we can go the significantly longer route of checking each ID and then having a switch statement and case for each of the 4 check boxes

Reply:
well I would either use tags in the actual code or set the text explicitly so it's modular